datafusion/core/tests/dataframe/dataframe_functions.rs

```diff
@@ -161,6 +161,181 @@ async fn test_fn_btrim_with_chars() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
```

Comment:
While adding a test for `coalesce` I realized the other functions in `core` weren't covered so I added new tests for them too (except for `get_field` which needs a map / struct and I was too lazy to add one to the test fixture